src/components/duration.rs

```diff
@@ -451,6 +452,225 @@ impl Duration {
         }
     }
 
+    #[inline]
+    pub fn round_v2<C: CalendarProtocol, Z: TzProtocol>(
+        &self,
+        increment: Option<RoundingIncrement>,
+        smallest_unit: Option<TemporalUnit>,
+        largest_unit: Option<TemporalUnit>,
+        rounding_mode: Option<TemporalRoundingMode>,
```

Comment:
Seeing how many optionals are here, maybe we should extract this into a `RoundingOptions` struct that implements `Default`.

Reply:
Yeah, I was thinking about doing this in a follow up, but it could be added to this as well.

Temporal spec refers to them as difference settings `13.53 GetDifferenceSettings`, but I don't hate `RoundingOptions` as a name. Any preference between `DifferenceSettings` vs. `RoundingOptions`

Reply:
I prefer `RoundingOptions` for consistency. If we need `DifferenceOptions` in some other place (the diff_instant operation maybe?), we can do a `type DifferenceOptions = RoundingOptions;` for it.

Reply:
From what I can tell they both essentially resolve to the same options but with different defaults and methodology. Updated them to a public `DifferenceSettings` and `RoundingOptions` that resolve to `ResolvedRoundingOptions`, let me know what you think!